### ISSUE TITLE
Partial scanning broken for template+imported types

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#6926]
Partial scanning is broken with a specific combination of template types and psalm types:

a.php:
```php
<?php

namespace Daniil\PsalmT;

/**
 * @template T as array
 */
class a {
    /**
     * @param T $t
     */
    public function take(array $t): void {
    }
}
```

b.php:
```php
<?php

namespace Daniil\PsalmT;

/**
 * @psalm-import-type T from c
 * 
 * @extends a<T>
 */
class b extends a {
}
```

c.php:
```php
<?php

/**
 * @psalm-type T = array{"test": int}
 */
class c {}
```

d.php:
```php
<?php

use Daniil\PsalmT\b;

$test = new b;
$test->take(['test' => 123]);
```

The following invalid issue is raised when scanning only `src/d.php`:

```
ERROR: InvalidArgument - src/d.php:6:13 - Argument 1 of Daniil\PsalmT\b::take expects type-alias(Daniil\PsalmT\c::T), array{test: 123} provided (see https://psalm.dev/004)
```

**Comment by danog** [comment]
Nevermind, the bug is still present in the original code, but the example I specified does not reproduce it.

**Comment by orklah** [comment]
If I understand this right, you're analyzing a file that has a psalm-import-type to another file that is not in the scope and it causes an error?

Can you paste the original error?

**Comment by danog** [comment]
Nevermind, it was just my extension messing with psalm ;)
